File: backend/src/main/java/com/example/demo/model/Message.java
package com.example.demo.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

@Entity
@Getter
@Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
@Table(name = "messages")
public class Message {

    @Id @GeneratedValue
    private Long id;

    @ManyToOne
    private Chat chat;

    @ManyToOne
    private ChatMember sender;

    private String content;

    private String fileUrl;

    private Instant timestamp = Instant.now();


}

File: frontend/models/Message.ts
import { ChatMember } from "./ChatMember";

export type MessageRequest = {
    chatId: number;
    senderUserId: number;
    content: string;
    fileUrl?: File
}

export type Message = {
  id: number;
  chatId: number;
  sender: ChatMember;
  content: string;
  fileUrl: string | null;
  timestamp: string; // ISO 8601 timestamp, e.g. "2025-10-24T10:15:30Z"
}


